email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,393 @@
+    }
+
+    /**
+     * Determine if the queue is empty
+     * 
+     * @return boolean   whether the queue is empty
+     */
+    public boolean isQEmpty() {
+        return handler.isEmpty();
+    }
+
+    /**
+     * Remove email from the head of the queue
+     * 
+     * @param msg   the Message variable to assign the removed message to
+     */
+    public void removeFromQ(EmailType msg) {
+        determineMsg(msg)[0] = handler.removeFirst();
+    }
+
+    /**
+     * Retrieve and return text from the specified email
+     * 
+     * @param msg   the message to read. Expressed by an integer value
+     * @return msgContent  the text from the specified email   
+     */
+    public String readEmail(EmailType msg) {
+        try {
+            Message mess = determineMsg(msg)[0];
+            System.out.println("---------------------------------");
+            System.out.println("From: " + mess.getFrom()[0]);
+            System.out.println("Subject: " + mess.getSubject());
+    
+            String msgContent = "";
+            response = "";
+            msgContent += writePart(mess);
+            System.out.println("Email reads:  \"" + msgContent + "\"");
+    
+            return msgContent;
+        } catch (MessagingException e) {
+            logger.error("Failed to fetch email body");
+            return "Error occured while reading email";
+        }
+        
+    }
+
+    /**
+     * Retrieve the body of an email
+     * 
+     * @param part   the email body to be retrieved
+     * @return response   the text of an email
+     */
+    public String writePart(Part part) {
+        try {
+            if (part.isMimeType("text/plain")) {
+                response += ((String) part.getContent());
+            } else if (part.isMimeType("multipart/*")) {
+                Multipart mp = (Multipart) part.getContent();
+                int count = mp.getCount();
+                for (int n = 0; n < count; n++) {
+                    String text = (writePart(mp.getBodyPart(n)));
+                }
+            } else if (part.isMimeType("message/rfc822")) {
+                response += writePart((Part) part.getContent());
+            }
+            return response;
+        } catch (MessagingException e) {
+            logger.error("Failed to fetch email body");
+            return "An error occured while processing your email.";
+        } catch (IOException e) {
+            logger.error("Failed to fetch email body");
+            return "An error occured while processing your email.";
+        }
+
+    }
+
+    /**
+     * Copy specified email from one folder to another
+     * 
+     * @param fromFolder   the folder email is currently located in
+     * @param toFolder   the folder email is going to be moved to
+     * @param msg   the message to be moved
+     * @return message that email was moved TODO
+     */
+    public String copyEmail(EmailType fromFolder, EmailType toFolder, EmailType msg) {
```

Comment:
Does this copy or move? The name and javadoc say one thing and the return message say another.